parser/src/fstring.rs

```diff
@@ -30,17 +30,39 @@ impl<'a> FStringParser<'a> {
         while let Some(ch) = self.chars.next() {
             match ch {
                 '!' if delims.is_empty() => {
-                    conversion = Some(match self.chars.next() {
-                        Some('s') => ConversionFlag::Str,
-                        Some('a') => ConversionFlag::Ascii,
-                        Some('r') => ConversionFlag::Repr,
-                        Some(_) => {
-                            return Err(InvalidConversionFlag);
+                    let x = self.chars.next();
+                    if let Some('=') = x {
+                        expression.push(ch);
+                        expression.push('=');
+                    } else {
+                        if expression.is_empty() {
+                            return Err(EmptyExpression);
                         }
-                        None => {
-                            break;
+
+                        conversion = Some(match x {
+                            Some('s') => ConversionFlag::Str,
+                            Some('a') => ConversionFlag::Ascii,
+                            Some('r') => ConversionFlag::Repr,
+                            Some(_) => {
+                                return Err(InvalidConversionFlag);
+                            }
+                            None => {
+                                break;
+                            }
+                        });
+
+                        match self.chars.peek() {
+                            Some('}') => {
+                                continue;
+                            }
+                            Some(_) => {
+                                return Err(ExpectedRbrace);
+                            }
+                            None => {
+                                break;
+                            }
```

Comment:
Same kinda thing here with `None`:
```py
>>> f"{a!r"
  File "<stdin>", line 1
SyntaxError: f-string: expecting '}'
```

Reply:
Turns out this could be simplified to just self.chars.peek() != Some(&'}' {}